spyder/app/tour.py

```diff
@@ -782,8 +783,11 @@ def set_data(self, title, content, current, image, run, frames=None,
         if run is None:
             self.button_run.setVisible(False)
         else:
-            self.button_run.setDisabled(False)
-            self.button_run.setVisible(True)
+            if sys.platform == "darwin":
+                self.button_run.setVisible(False)
+            else:
+                self.button_run.setDisabled(False)
+                self.button_run.setVisible(True)

```

Comment:
Instead of hiding `button_run`, let's simply remove it. I think it won't be possible for us to show it in the future due to the problems it introduces in macOS.

Reply:
I don't agree with removing it in all platforms for the sake of one (see above), but if we do, we should remove the `run` key from the get_tour function (which prevents the button from being shown or code being run from this tour) and revert this change, rather than removing an otherwise working feature completely (and breaking the other tours too). We were almost able to fix/work around this bug but not quite, so it might well be possible to resolve in the future either with action ourselves or with dependency (Qt, etc) updates, and given the macOS transition to ARM, removal of software development features and shift away from the professional space, we may not support it in the future so I don't think it makes sense to throw out this otherwise useful feature completely.

Reply:
Ok, fair enough.

Reply:
Per our consensus above. @juanis2112 if you have a better idea for the tooltip text, I'll defer to whatever you come up with.

```suggestion
            self.button_run.setVisible(True)
            self.button_run.setDisabled(sys.platform == "darwin")
            if sys.platform == "darwin":
                self.button_run.setToolTip("Not available on macOS")
```